package/MDAnalysis/coordinates/TRJ.py

```diff
@@ -458,74 +463,164 @@ class NCDFReader(base.ReaderBase):
-                "with a time unit of picoseconds and not {0}.".format(
-                    self.trjfile.variables['time'].units))
-        if self.trjfile.variables['coordinates'].units.decode('utf-8') != "angstrom":
-            raise NotImplementedError(
-                "NETCDFReader currently assumes that the trajectory was written "
-                "with a length unit of Angstroem and not {0}.".format(
-                    self.trjfile.variables['coordinates'].units))
-        if hasattr(self.trjfile.variables['coordinates'], 'scale_factor'):
-            raise NotImplementedError("scale_factors are not implemented")
-        if n_atoms is not None and n_atoms != self.n_atoms:
-            raise ValueError("Supplied n_atoms ({0}) != natom from ncdf ({1}). "
-                             "Note: n_atoms can be None and then the ncdf value "
-                             "is used!".format(n_atoms, self.n_atoms))
+        verify_units(self.trjfile.variables['time'].units, ('picosecond',))
+        verify_units(self.trjfile.variables['coordinates'].units,
+                     ('angstrom',))
+
+        # Check for scale_factor attributes for all data variables and
+        # store this to multiply through later (Issue #2323)
+        self.scale_factors = {'time': 1.0,
+                              'cell_lengths': 1.0,
+                              'cell_angles': 1.0,
+                              'coordinates': 1.0,
+                              'velocities': 1.0,
+                              'forces': 1.0}
+
+        for variable in self.trjfile.variables:
+            if hasattr(self.trjfile.variables[variable], 'scale_factor'):
+                if variable in self.scale_factors:
+                    scale_factor = self.trjfile.variables[variable].scale_factor
+                    self.scale_factors[variable] = scale_factor
+                else:
+                    errmsg = ("scale_factors for variable {0} are "
+                              "not implemented".format(variable))
+                    raise NotImplementedError(errmsg)
 
         self.has_velocities = 'velocities' in self.trjfile.variables
+        if self.has_velocities:
+            verify_units(self.trjfile.variables['velocities'].units,
+                         ('angstrom/picosecond',))
+
         self.has_forces = 'forces' in self.trjfile.variables
+        if self.has_forces:
+            verify_units(self.trjfile.variables['forces'].units,
+                         ('kilocalorie/mole/angstrom',))
 
         self.periodic = 'cell_lengths' in self.trjfile.variables
+        if self.periodic:
+            verify_units(self.trjfile.variables['cell_lengths'].units,
+                         ('angstrom',))
+            # Accept plural as this is what MDA writes as of 0.19.2
+            verify_units(self.trjfile.variables['cell_angles'].units,
+                         ('degree', 'degrees'))
```

Comment:
How about we accept "degrees" until 1.0 but issue a deprecation warning right now.

Can we 
- change degrees to degree when we encounter it
- warn the user when we correct degrees
- issue a deprecation warning that from 1.0 onwards, files with cell_angles in degrees will raise an error
- only write degree

Reply:
Should be addressed in 54da919.